lib/base_service.ts

```diff
@@ -252,13 +298,19 @@ export class BaseService {
    * @returns {Credentials}
    */
   private getCredentialsFromBluemix(vcapServicesName: string): Credentials {
-    let _credentials: Credentials;
+    let credentials: Credentials;
+    let temp: any;
     if (this.name === 'visual_recognition') {
-      _credentials = vcapServices.getCredentials('watson_vision_combined');
+      temp = vcapServices.getCredentials('watson_vision_combined');
     } else {
-      _credentials = vcapServices.getCredentials(vcapServicesName);
+      temp = vcapServices.getCredentials(vcapServicesName);
     }
-    return _credentials;
+    // convert an iam apikey to use the identifier iam_apikey
+    if (temp.apikey && temp.iam_apikey_name) {
+      temp.iam_apikey = temp.apikey;
+      delete temp.apikey;
+    }
+    credentials = temp;
+    return credentials;
```

Comment:
Was a little confused to see `temp` used as the variable name here. Although I can see why, given that `credentials` and `vcapServices` are both already defined.